unison-cli/src/Unison/Codebase/TranscriptParser.hs

```diff
@@ -316,8 +321,8 @@ run dir stanzas codebase runtime sbRuntime config ucmVersion baseURL = UnliftIO.
                     [] -> awaitInput loopState
                     args -> do
                       output ("\n" <> show p <> "\n")
-                      let currentRoot = Branch.head (loopState ^. #root)
-                      case parseInput currentRoot curPath (loopState ^. #numberedArgs) patternMap args of
+                      let getRoot = fmap Branch.head . atomically $ readTMVar (loopState ^. #root)
```

Comment:
Could use getRootBranch

Reply:
Do you mean the `Cli` version or something else?

This isn't in the Cli monad 😢